lib/java_buildpack/framework/introscope_agent.rb

```diff
@@ -80,12 +66,27 @@ def agent_jar
         @droplet.sandbox + 'Agent.jar'
       end
 
-      def agent_manager(credentials)
-        agent_manager = ssl?(credentials) ? 'https://' : 'http://'
-        agent_manager += host_name(credentials)
+      def add_url(credentials, java_opts)
+        agent_manager = url(credentials)
 
-        port = port(credentials)
-        port ? "#{agent_manager}:#{port}" : agent_manager
+        host, port, socket_factory = parse_url(agent_manager)
+        java_opts.add_system_property('agentManager.url.1', agent_manager)
+        java_opts.add_system_property('introscope.agent.enterprisemanager.transport.tcp.host.DEFAULT', host)
```

Comment:
The agentManager.url.1 property is very familiar with our customers, and has been in use in our on premise product. We have received feedback that our customers will want to continue using the same property while configuring the buildpack as well. Hence, we want to stick to using the "url" as the input. Host, port, and socketfactory properties are for much older agents. They are meant for backwards compatibility. 

To summarize, I have set both host and agentManager.url.1 because host is meant to support the older agent, while the agentManager.url.1. is meant to support newer agents.